instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/r2dbc/R2dbcInstrumentingPostProcessor.java

```diff
@@ -17,20 +19,25 @@
 class R2dbcInstrumentingPostProcessor implements BeanPostProcessor {
 
   private final ObjectProvider<OpenTelemetry> openTelemetryProvider;
-  private final boolean statementSanitizationEnabled;
+  private final ObjectProvider<ConfigProperties> configPropertiesProvider;
 
   R2dbcInstrumentingPostProcessor(
-      ObjectProvider<OpenTelemetry> openTelemetryProvider, boolean statementSanitizationEnabled) {
+      ObjectProvider<OpenTelemetry> openTelemetryProvider,
+      ObjectProvider<ConfigProperties> configPropertiesProvider) {
     this.openTelemetryProvider = openTelemetryProvider;
-    this.statementSanitizationEnabled = statementSanitizationEnabled;
+    this.configPropertiesProvider = configPropertiesProvider;
   }
 
   @Override
   public Object postProcessAfterInitialization(Object bean, String beanName) {
     if (bean instanceof ConnectionFactory && !ScopedProxyUtils.isScopedTarget(beanName)) {
       ConnectionFactory connectionFactory = (ConnectionFactory) bean;
       return R2dbcTelemetry.builder(openTelemetryProvider.getObject())
-          .setStatementSanitizationEnabled(statementSanitizationEnabled)
+          // there is no instrumentation-specific property, so we use the common one
+          .setStatementSanitizationEnabled(
+              InstrumentationConfigUtil.isStatementSanitizationEnabled(
+                  configPropertiesProvider.getObject(),
+                  "otel.instrumentation.common.db-statement-sanitizer.enabled"))
```

Comment:
checks `otel.instrumentation.common.db-statement-sanitizer.enabled` twice

Reply:
yes - because agent is only using this prop

Reply:
(?)
```suggestion
          .setStatementSanitizationEnabled(
              InstrumentationConfigUtil.isStatementSanitizationEnabled(
                  configPropertiesProvider.getObject(),
                  "otel.instrumentation.r2dbc.statement-sanitizer.enabled"))
```

Reply:
are you saying that we should also do this change in the agent?

Reply:
I like the idea of adding dedicated flag. Added one for the javaagent instrumentation in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11384